Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp

```diff
@@ -461,6 +461,15 @@ static std::optional<uint64_t> convertToUint64(JSC::JSValue jsValue)
     return std::nullopt;
 }
 
+static std::optional<double> convertToDouble(JSC::JSValue jsValue)
```

Comment:
Probably slightly better to use JSValueToNumber instead. It will handle a few types not handled here, and provide a more specific error message if conversion fails.

Reply:
The current IPCTestingAPI.cpp is written mostly with the pattern that the VM lock is held in the entering function and all the calls are to the internal APIs, not the external API. Also the exception message is worse in the sense that it wouldn't say which parameter was incorrect. Leaving as is as it is more consistent with the surrounding code -- if the public API should be used all around, the code should be rewritten with that goal in mind.